xaitk_saliency/impls/perturb_image/occlusion.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict, Any, Iterable, Tuple, List
+
+import PIL.Image
+import numpy as np
+
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+from xaitk_saliency.utils.masking import generate_block_masks
+
+
+class OcclusionBasedPerturb (PerturbImage):
+    """
+    Produce image perturbations based on hard, block-y occlusion areas as
+    generated by sliding a window of a configured size.
+
+    For multi-channel input, all channels are occluded at the same time or not
+    at all.
+    For example, if the input image is an RGB image, for regions masked, R, G
+    and B channels will be masked at the same time.
+    If the image was a 2x2 white image, one mask of just the upper-left pixel
+    might be as follows::
+
+        >>> img = (
+        ...    [[[255, 255, 255], [255, 255, 255]],
+        ...     [[255, 255, 255], [255, 255, 255]]]
+        ... )
+        >>> img_mask_example = (
+        ...     [[[  0,   0,   0], [255, 255, 255]],
+        ...      [[255, 255, 255], [255, 255, 255]]]
+        ... )
+
+    :param window_size: The block window size.
+    :param stride: The sliding window striding step.
+    """
+
+    def __init__(self, window_size: int = 50, stride: int = 20):
+        self.window_size = int(window_size)
+        self.stride = int(stride)
```

Comment:
Do we plan to only support square window sizes and symmetric strides? I imagine this covers 95% of use cases, but I can potentially image where non-square occluders (e.g. rectangles) or non-symmetric strides (i.e. different in x,y) might be useful. I see other libraries either allow ints or tuples of ints to account for this.

Reply:
I don't want to get too far into the weeds of possible use-cases, but this seems sensible and simple enough. I'll take a stab at adding that here, circling back if the change is more of a sink than expected.

Reply:
Great point, I have come across use cases with non - square occluders generally used as synthetic distractors to strengthen training.

Reply:
The new update should have tuple inputs for window and stride and an added unit-test that tests a rectangular window/stride example.

Reply:
Looks good to me- thanks for making the change.